### PR TITLE
[aks-preview] Support updating an existing AAD-Integrated cluster to the new AKS-managed AAD experience

### PR DESCRIPTION
**Pull request by bingosummer**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
aks-preview

File: src/aks-preview/setup.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from codecs import open as open1
 from setuptools import setup, find_packages
 
-VERSION = "0.4.51"
+VERSION = "0.4.52"
```
Comment (bingosummer) [inline src/aks-preview/setup.py:11]: Will update the version

**Comment by bingosummer** [comment]
> Hi @bingosummer , is this PR related to #1887 ? #1887 is waiting for 2020-06-01 version published. Does this also have that dependency ?

This PR doesn't need the 2020-06-01 version. But it also need to wait for this week's release. I'll update the PR then.

**Comment by bingosummer** [comment]
@arrownj Ready for review.

@weinong @karataliu @zqingqing1 Please help review. Thanks!

**Comment by arrownj** [comment]
Hi @bingosummer , the code is good to me,  please ping me when you think it is ready to merge. And don't forget solve the code conflict.

**Comment by bingosummer** [comment]
@arrownj It's ready to merge.

File: src/aks-preview/azext_aks_preview/_help.py [hunk 2]
```diff
@@ -389,6 +394,10 @@
         text: az aks update -g MyResourceGroup -n MyManagedCluster --api-server-authorized-ip-ranges ""
       - name: Restrict apiserver traffic in a kubernetes cluster to agentpool nodes.
         text: az aks update -g MyResourceGroup -n MyManagedCluster --api-server-authorized-ip-ranges 0.0.0.0/32
+      - name: Update a AKS-managed AAD cluster with tenant ID or admin group object IDs.
+        text: az aks update -g MyResourceGroup -n MyManagedCluster --aad-admin-group-object-ids <id-1,id-2> --aad-tenant-id <id>
```
Comment (zqingqing1) [inline src/aks-preview/azext_aks_preview/_help.py:398]: shouldn't it have --enable-aad?

File: src/aks-preview/azext_aks_preview/custom.py [hunk 3]
```diff
@@ -1155,6 +1156,7 @@ def aks_update(cmd,     # pylint: disable=too-many-statements,too-many-branches,
        not update_pod_security and \
        not update_lb_profile and \
        not uptime_sla and \
+       not enable_aad and \
        not update_aad_profile:
         raise CLIError('Please specify "--enable-cluster-autoscaler" or '
                        '"--disable-cluster-autoscaler" or '
```
Comment (zqingqing1) [inline src/aks-preview/azext_aks_preview/custom.py:1162]: should add "--enable-aad" in the error msg

File: src/aks-preview/azext_aks_preview/_help.py (hunk 2 above)
Comment (bingosummer) [reply src/aks-preview/azext_aks_preview/_help.py:398]: This example is to update a cluster whose managed AAD is already enabled. So it should not have --enable-aad.

There's another example below to enable AAD for the AADv1 cluster. This one should have --enable-aad.

**Comment by palma21** [comment]
any blockers to merge this?

**Comment by bingosummer** [comment]
> any blockers to merge this?

No blockers. The AKS RP changes have rolled out.

@arrownj Please help merge the PR.

**Pull request merged**